src/rpc/net.cpp

```diff
@@ -121,73 +184,82 @@ UniValue getpeerinfo(const JSONRPCRequest& request)
-                heights.push_back(height);
-            }
-            obj.push_back(Pair("inflight", heights));
+        CNodeStats stats;
+
+        int node_id = request.params[0].get_int();
+        if (!g_connman->GetNodeStats(node_id, stats)) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Node id %u not found", node_id));
         }
-        obj.push_back(Pair("whitelisted", stats.fWhitelisted));
+        NodeStatsToJSON(stats, obj);
+        ret.push_back(obj);
+    } else {
+        std::vector<CNodeStats> vstats;
+        g_connman->GetNodeStats(vstats);
+
+        for (const CNodeStats& stats : vstats) {
+            UniValue obj(UniValue::VOBJ);
+            NodeStatsToJSON(stats, obj);
 
-        UniValue sendPerMsgCmd(UniValue::VOBJ);
-        for (const mapMsgCmdSize::value_type &i : stats.mapSendBytesPerMsgCmd) {
-            if (i.second > 0)
-                sendPerMsgCmd.push_back(Pair(i.first, i.second));
+            ret.push_back(obj);
         }
-        obj.push_back(Pair("bytessent_per_msg", sendPerMsgCmd));
+    }
 
-        UniValue recvPerMsgCmd(UniValue::VOBJ);
-        for (const mapMsgCmdSize::value_type &i : stats.mapRecvBytesPerMsgCmd) {
-            if (i.second > 0)
-                recvPerMsgCmd.push_back(Pair(i.first, i.second));
+    return ret;
+}
+
+UniValue updatepeer(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 3)
+        throw std::runtime_error(
+            "updatepeer\n"
+            "\nUpdate settings for a network peer (debug method).\n"
+            "\nReturns updated information about the peer. See getpeerinfo() for the format of the returned object."
+            "\nArguments:\n"
+            "1. \"id\"                (string, required) The peer id (see getpeerinfo for a list of peers with their ids)\n"
+            "2. \"whitelisted\"       (bool, optional) whether the peer is whitelisted\n"
+            "3. \"manual_connection\" (bool, optional) whether the peer was manually added\n"
+            "\nExamples:\n"
+            + HelpExampleCli("updatepeer", "0 true")
+            + HelpExampleRpc("updatepeer", "0, true")
+        );
+
+    if (!g_connman) {
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+    }
+
+    LOCK(cs_main);
+
+    RPCTypeCheckArgument(request.params[0], UniValue::VNUM);
+    int node_id = request.params[0].get_int();
+
+    CNodeStats stats;
+    if (!g_connman->GetNodeStats(node_id, stats)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Node id %u not found", node_id));
+    }
+
+    if (!request.params[1].isNull()) {
+        RPCTypeCheckArgument(request.params[1], UniValue::VBOOL);
+        if (!g_connman->SetWhitelisted(node_id, request.params[1].isTrue())) {
+            throw JSONRPCError(RPC_MISC_ERROR, "Failed to update peer whitelisting");
```

Comment:
See my above comment, by fetching once the node at the start of this method, you can remove all those `if` as you already know if the CNode with such id already exist.